ec/src/wnaf.rs

```diff
@@ -0,0 +1,43 @@
+extern crate alloc;
+use crate::ProjectiveCurve;
+use alloc::vec::Vec;
+
+// Computes scalar multiplication of a preprocessed group element by a scalar in windowed non-adjacent form (wNAF).
+// `scalar_wnaf` is expected in its NAF form which can be obtained by using `find_wnaf` on the underlying BigInt.
+// `table` holds pre-computed multiples of the group element; it can be calculated using `wnaf_table`.
+pub fn wnaf_mul<G: ProjectiveCurve>(table: &[G], scalar_wnaf: &[i64]) -> G {
```

Comment:
Should we rename this to `wnaf_mul_with_table`, and add a function `wnaf_mul` that works roughly like:
```rust
pub fn wnaf_mul<G: ProjectiveCurve>(element: &G, scalar_wnaf: &[i64]) -> G {
	// calculate table size
	let window_size = something();
	let table = wnaf_table(*element, window_size);
	wnaf_mul_with_table(&table, scalar_wnaf)
}
```

Reply:
I don't know any particular way of deciding the window size. Do you have anything specific in mind?

Reply:
Hmm it seems that right now the `wnaf_mul` passes tests because the output of `BigInteger::find_wnaf` assume a window size of just 2, and so we never access something that's not in the table. However, if we change the table size to 1, the tests should fail because `wnaf_mul` accesses a multiple that's not in the table. 

@ggkitsas Do you mind giving me access to your branch so that I can push some fixes for this?

Reply:
Ahh I see, @Pratyush invitation sent